File: src/main/java/com/github/xumeng/java/problems/easy/problem0226/Solution0226.java
package com.github.xumeng.java.problems.easy.problem0226;

import com.github.xumeng.java.common.datastructure.TreeNode;

public class Solution0226 {

    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        if (root.left != null) {
            invertTree(root.left);
        }
        if (root.right != null) {
            invertTree(root.right);
        }
        TreeNode node = root.left;
        root.left = root.right;
        root.right = node;
        return root;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0226/Solution0226.kt
package com.github.xumeng.kotlin.problems.easy.problem0226

import com.github.xumeng.kotlin.common.datastructure.TreeNode

class Solution0226 {

    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null) {
            return null
        }
        if (root.left != null) {
            invertTree(root.left)
        }
        if (root.right != null) {
            invertTree(root.right)
        }
        val node: TreeNode? = root.left
        root.left = root.right
        root.right = node
        return root
    }

}